Specs/Core/loadKTX2Spec.js

```diff
@@ -664,198 +276,90 @@ describe("Core/loadKTX2", function () {
+      expect(promise).toBeDefined();
+      return promise.then(function (resolvedValue) {
+        expect(resolvedValue).toBeDefined();
+        expect(resolvedValue.width).toEqual(4);
+        expect(resolvedValue.height).toEqual(4);
+        expect(
+          PixelFormat.isCompressedFormat(resolvedValue.internalFormat)
+        ).toEqual(false);
+        expect(resolvedValue.bufferView).toBeDefined();
+      });
+    });
   });
 
   it("returns a promise that resolves to an uncompressed texture containing all mip levels of the original texture", function () {
-    var testUrl = "http://example.invalid/testuri";
-    var promise = loadKTX2(testUrl);
-
-    expect(promise).toBeDefined();
-
-    var resolvedValue;
-    var rejectedError;
-    promise.then(
-      function (value) {
-        resolvedValue = value;
-      },
-      function (error) {
-        rejectedError = error;
-      }
-    );
-
-    expect(resolvedValue).toBeUndefined();
-    expect(rejectedError).toBeUndefined();
-
-    var response = validUncompressedMipmap.buffer;
-    fakeXHR.simulateLoad(response);
-    expect(resolvedValue).toBeDefined();
-    expect(resolvedValue.length).toEqual(3);
-    expect(resolvedValue[0].width).toEqual(4);
-    expect(resolvedValue[0].height).toEqual(4);
-    expect(
-      PixelFormat.isCompressedFormat(resolvedValue[0].internalFormat)
-    ).toEqual(false);
-    expect(resolvedValue[0].bufferView).toBeDefined();
-    expect(rejectedError).toBeUndefined();
+    var resource = Resource.createIfNeeded("./Data/Images/Green4x4Mipmap.ktx2");
+    var loadPromise = resource.fetchArrayBuffer();
+    return loadPromise.then(function (buffer) {
+      var promise = KTX2Transcoder.transcode(buffer, { pvrtc: true });
+      expect(promise).toBeDefined();
+      return promise.then(function (resolvedValue) {
+        expect(resolvedValue).toBeDefined();
+        expect(resolvedValue.length).toEqual(3);
+        var dims = [4, 2, 1];
+        for (var i = 0; i < resolvedValue.length; ++i) {
+          expect(resolvedValue[i].width).toEqual(dims[i]);
+          expect(resolvedValue[i].height).toEqual(dims[i]);
+          expect(resolvedValue[i].bufferView).toBeDefined();
+        }
+        expect(
+          PixelFormat.isCompressedFormat(resolvedValue.internalFormat)
+        ).toEqual(false);
+      });
+    });
   });
 
-  it("returns a promise that resolves to a compressed texture when the request loads", function () {
-    var testUrl = "http://example.invalid/testuri";
-    var promise = loadKTX2(testUrl);
-
-    expect(promise).toBeDefined();
-
-    var resolvedValue;
-    var rejectedError;
-    promise.then(
-      function (value) {
-        resolvedValue = value;
-      },
-      function (error) {
-        rejectedError = error;
-      }
+  it("returns a promise that resolves to a compressed texture", function () {
+    var resource = Resource.createIfNeeded(
+      "./Data/Images/Green4x4Compressed.ktx2"
```

Comment:
Currently this points to an ETC1S ktx2 file I made. Should there be UASTC versions of the regular + mipmap file?